assets/javascripts/modules/form/email-check.js

```diff
@@ -0,0 +1,45 @@
+define(['bean',
+    'utils/ajax',
+    'utils/text',
+], function (bean, ajax, textUtils) {
+
+    'use strict';
+
+    function init() {
+
+        var YOUR_DETAILS_SUBMIT_ELEM = document.querySelector('.js-checkout-your-details-submit');
+        var EMAIL_ID_INPUT_ELEM      = document.querySelector('.js-checkout-email');
+        var USER_SIGNED_IN           = document.querySelector('data-user-signedin="true"]')
+
+        // do nothing if the user is already signed in
+        if (USER_SIGNED_IN) { return; }
+
+        ajax.init({page: {ajaxUrl: ''}});
+
+        bean.on(YOUR_DETAILS_SUBMIT_ELEM, 'click', function (event) {
+            if (EMAIL_ID_INPUT_ELEM) {
+                var email = textUtils.removeWhitespace(EMAIL_ID_INPUT_ELEM.value);
+
+                if (email.length) {
+                    ajax({
+                        url: '/checkout/check-identity?email=' + email
+                    }).then(function (response) {
+                        if (!response.emailInUse) {
+                            bean.fire(YOUR_DETAILS_SUBMIT_ELEM, "submit");
+                        } else {
+                            alert("Your email is already in use! Please sign in or use another email address");
+                        }
+                    }).fail(function (_, msg) {
+                        console.error("Error reaching endpoint /checkout/check-identity:", msg);
```

Comment:
I'm guessing this `console` and the above `alert` are just for debugging purposes at the moment? Will need to remove them afterwards.